### PR TITLE
Docs/partial rendering

### PR DESCRIPTION
**Pull request by Francois-Esquire**
Recipe/guide for Partial Rendering with sample walk through.

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation (adding or updating documentation)
- [ ] Dependency update

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [ ] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using One App?

Docs to using partial rendering.

### CODE REVIEW

File: docs/recipes/Partial-Rendering.md [hunk 1]
```diff
@@ -4,18 +4,172 @@
...
+
+// ...
+```
+
+Since partial rendering is configurable only during server side rendering,
+a partial render can source props from the request body via initial state building.
+The Holocron config adds a reducer which stores the props from the request body.
+
+```js
 // ...
```
Comment (JAdshead) [inline docs/recipes/Partial-Rendering.md:95]: I'm not sure if its clear enough that this is a continuation.

```suggestion
```js
// continued RootModule

```

File: docs/recipes/Partial-Rendering.md [hunk 2]
```diff
@@ -4,18 +4,172 @@
...
+  const { params: { moduleName } } = props;
+  return (
+    <RenderModule
+      moduleName={moduleName}
+      props={props}
+    />
+  );
+}
+
+export default function createPartialRoute(getPartialRenderProps) {
```
Comment (JAdshead) [inline docs/recipes/Partial-Rendering.md:61]: should we start with a simpler example of a plain `ModuleRoute` and then suggest this as a possibility if they are creating multiple partial routes.

**Review by JAdshead — CHANGES_REQUESTED** [review]

File: docs/recipes/Partial-Rendering.md (hunk 1 above)
Comment (Francois-Esquire) [reply docs/recipes/Partial-Rendering.md:95]: Good suggestion, I'll update with this

File: docs/recipes/Partial-Rendering.md (hunk 2 above)
Comment (Francois-Esquire) [reply docs/recipes/Partial-Rendering.md:61]: I agree with this point, I'll update

**Review by infoxicator — COMMENTED** [review]
I think we should maybe focus this example on a real use case like rendering an email? and also expand on the need for `<dangerously-return-only-doctype />` and also an example of why that is required?

File: docs/recipes/Partial-Rendering.md [hunk 3]
```diff
@@ -4,18 +4,172 @@
 
 # Partial Rendering
 
+In this recipe, we will go over the setup required to start using partial rendering with One App,
```
[the remark this answers is not in the document]
Comment (Francois-Esquire) [reply docs/recipes/Partial-Rendering.md]: I've updated this a bit but I've kept the suggested sentence (5809765)

**Comment by Francois-Esquire** [comment]
> I think we should maybe focus this example on a real use case like rendering an email? and also expand on the need for `<dangerously-return-only-doctype />` and also an example of why that is required?

@infoxicator I've updated to include an email example using `dangerously-return-only-doctype` (5809765)

**Review by JAdshead — DISMISSED** [review]

File: docs/recipes/Partial-Rendering.md [hunk 4]
```diff
@@ -4,24 +4,324 @@
...
+  return (
+    <dangerously-return-only-doctype
+      dangerouslySetInnerHTML={{ __html: html }}
+    />
+  );
+}
+```
+
+Do keep in mind that partial rendering is meant to be performed server-side, we should not attempt
+this example above in the browser in normal circumstances.
```
Comment (mtomcal) [inline docs/recipes/Partial-Rendering.md:309]:
```suggestion
this example above in the browser under normal circumstances.
```

**Review by mtomcal — CHANGES_REQUESTED** [review]
Small nit. Overall quite well written.

**Review by mtomcal — APPROVED** [review]

**Review by JAdshead — APPROVED** [review]

**Pull request merged**